### PR TITLE
Fix missing exit code on build script failure

### PR DESCRIPTION
**Pull request by xzyfer**
Currently the build command always exits with a `0` code when
a compilation error is encountered which prevents compilation 
failures failing builds in CI environments.

>The `err` object will not include compilation errors and those 
>must be handled separately using `stats.hasErrors()` [...]. The 
>`err` object will only contain webpack-related issues, such as
>misconfiguration, etc.
>[webpack node API documentation][3]

This PR sets the [exit code][1] to 1 when an error is encountered 
but lets the process exit naturally rather than a force exit, which 
[can be problematic][2] when async process are in play.

This has benefits over for exiting the process i.e. not exiting when
watching. It also doesn't require adding additional or cli flags.

Additionally this preserves non-zero exit codes if they've already
been set to preserve BC.

The specific implementation is inspired by Webpack's [Node API 
error handling documentation][4] with considerations made for
maintaining BC.

[1]: https://nodejs.org/api/process.html#process_process_exitcode
[2]: https://nodejs.org/api/process.html#process_process_exit
[3]: https://webpack.js.org/api/node/#webpack-
[4]: https://webpack.js.org/api/node/#error-handling

### CODE REVIEW

**Comment by ctrlplusb** [comment]
Zing, great catch @xzyfer!

**Review by ctrlplusb — APPROVED** [review]

**Comment by ctrlplusb** [comment]
@xzyfer would you mind rebasing this against `next`?  I am ready to merge but would like it to go into `next` as we are queuing up a few big updates there.

**Comment by xzyfer** [comment]
Done :)

**Comment by ctrlplusb** [comment]
Hi @xzyfer this PR is still showing a lot of the changes that were already added to `next`. I think it might be easier for you to create a new branch off of `next` and then create a new PR with your change here:
https://github.com/ctrlplusb/react-universally/pull/523/files#diff-5dbb54beea22fab8ea3a4af3ec9b96a6

Otherwise I am happy to apply the code change directly to `next` for you.

Let me know :)

**Comment by xzyfer** [comment]
Rebased on next

**Comment by ctrlplusb** [comment]
Perfect thanks 🤘

File: internal/scripts/build.js [hunk 1]
```diff
@@ -26,9 +26,18 @@ Object.keys(config('bundles'))
     const compiler = webpack(webpackConfigFactory({ target: bundleName, optimize }));
     compiler.run((err, stats) => {
       if (err) {
-        console.error(err);
+        console.error(err.stack || err);
+        if (err.details) {
+          console.error(err.details);
+        }
+        process.exitCode = process.exitCode || 1;
```
Comment (ctrlplusb) [inline internal/scripts/build.js]: Sorry for my ignorance but could you please explain the reasoning for setting the process exit code?

**Comment by ctrlplusb** [comment]
Should we call process.exit with the exit code explicitly?

**Comment by xzyfer** [comment]
The intent was to not override a non-zero exit code igniter was already set
by something else.

**Comment by ctrlplusb** [comment]
Yeah, sorry I realised this after I wrote the comment, but should we not just call `process.exit` with the error code immediately rather than set the `process.exitCode`?

**Comment by xzyfer** [comment]
This docs suggested force exiting [may not always be safe][1]. Given that the guard here prevents extra work happening after the exitcode is set it seemed fine to just let the process terminate naturally.

[1]: https://nodejs.org/api/process.html#process_process_exit_code

**Comment by ctrlplusb** [comment]
TIL! 👍

**Review by ctrlplusb — APPROVED** [review]

**Pull request merged**